### PR TITLE
Webchat styles styled components

### PR DESCRIPTION
**Pull request by imOscarCrespo**
In this PR I did some changes about:
- Styled components: I've used the library styled-components for updating the way they were defined, there are still components to do.
- Persistent menu: in our webchat.jsx now we can recieve from a bot a persistent menu. It's a must  for Hp, but we can use it for other clients
- Credit cards and phone numbers: Hp asked us to block when user input was a phone number or credit card.

### CODE REVIEW

File: packages/botonic-react/src/components/carousel.jsx [hunk 1]
```diff
@@ -1,8 +1,22 @@
 import React from 'react'
+import styled from 'styled-components'
 
 import { Message } from './message'
 import { isBrowser, isNode } from '@botonic/core'
 
+const Box = styled.div`
```
Comment (dpinol) [inline packages/botonic-react/src/components/carousel.jsx]: cool, I didn't know this syntax

File: packages/botonic-react/src/components/carousel.jsx [hunk 2]
```diff
@@ -16,33 +30,17 @@ export const Carousel = props => {
...
+      </Box>
     )
   }
   return (
     <Message
       style={{ maxWidth: '90%', padding: 0 }}
       blob={false}
       json={serialize(props)}
-      type='carousel'
+      type="carousel"
```
Comment (dpinol) [inline packages/botonic-react/src/components/carousel.jsx]: did you get an error with single quotes? @Khaeshah shared some prettier config to use " on jsx

File: packages/botonic-react/src/components/message.jsx [hunk 3]
```diff
@@ -77,6 +86,20 @@ export const Message = props => {
     return isFromUser() ? webchatState.theme.brandColor : '#F1F0F0'
   }
 
+  const getFontColor = props => {
+    let fontColorUser = '#ffffff'
```
Comment (dpinol) [inline packages/botonic-react/src/components/message.jsx]: thanks for trying the refactor! 👍

File: packages/botonic-react/src/webchat/webchatDev.jsx [hunk 4]
```diff
@@ -111,21 +116,21 @@ export const WebchatDev = forwardRef((props, ref) => {
             >
               <label style={{ flex: 'none', marginBottom: 6 }}>
                 <input
-                  type='checkbox'
-                  name='toggleSessionView'
+                  type="checkbox"
```
Comment (dpinol) [inline packages/botonic-react/src/webchat/webchatDev.jsx]: same, jsxSingleQuote

**Comment by imOscarCrespo** [comment]
> why did you remove them?

idk why it is not in this code, but i did a push with the changes

**Comment by vanbasten17** [comment]
Another thing that could be interesting. We can destructure the props in the same line we declare the parameters props => {prop1, prop2, ...}. I think this is useful when we are on a specific class to know what we have available to work with.
```javascript
const Test = ({ prop1, prop2 }) => {
  return <h1>{prop1}</h1>
};

function App() {
  return (
    <div className="App">
      <Test prop1={"hola"} />
    </div>
  );
}
```

**Pull request merged**